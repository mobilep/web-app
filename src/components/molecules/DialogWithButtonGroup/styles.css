.DialogWithButtonGroup {
	padding: 130px 94px;
	flex-direction: column;

	/* Fix for Safari dialog */
	width: 570px;
	height: 350px;
}

.DialogWithButtonGroup-content {
	margin-bottom: 60px;
}

.DialogWithButtonGroup-buttonGroup button {
	font-size: 16px;
	margin: 0 5px;
	min-width: 150px;
}

@media (max-width: 768px) {
	.DialogWithButtonGroup-content {
		font-size: 20px;
	}

	.DialogWithButtonGroup {
		width: 90%;
		padding-left: 24px;
		padding-right: 24px;
		padding-top: 64px;
		padding-bottom: 49px;
	}

	.DialogWithButtonGroup-buttonGroup button {
		min-width: 126px;
	}

	.DialogWithButtonGroup-buttonGroup {
		display: flex;
	}
}
