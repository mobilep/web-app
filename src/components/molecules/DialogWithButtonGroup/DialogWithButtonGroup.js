import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import { Dialog, Button } from '../../atoms';

import './styles.css';

class DialogWithButtonGroup extends Component {

	static propTypes = {
		actionButtonDisabled: PropTypes.bool,
		actionButtonLabel: PropTypes.string,
		cancelButtonLabel: PropTypes.string,
		children: PropTypes.any,
		className: PropTypes.string,
		forwardedRef: PropTypes.any,
		onActionButtonClick: PropTypes.func,
		onCancelButtonClick: PropTypes.func,
		onClose: PropTypes.func,
	}

	static defaultProps = {
		onActionButtonClick: () => {},
		onCancelButtonClick: () => {},
	}

	dialogRef = this.props.forwardedRef || React.createRef();

	handleCancelButtonClick = () => {
		this.dialogRef.current.close();
		this.props.onCancelButtonClick();
	};

	render () {
		const {
			actionButtonLabel,
			cancelButtonLabel,
			children,
			className,
			onActionButtonClick,
			actionButtonDisabled,
			onClose,
		} = this.props;
		return (
			<Dialog
				onClose={onClose}
				className={classNames('DialogWithButtonGroup', className)}
				ref={this.dialogRef}
			>
				<div className='DialogWithButtonGroup-content'>{children}</div>
				<div className='DialogWithButtonGroup-buttonGroup'>
					<Button color='grey' onClick={this.handleCancelButtonClick}>{cancelButtonLabel}</Button>
					<Button disabled={actionButtonDisabled} onClick={onActionButtonClick}>{actionButtonLabel}</Button>
				</div>
			</Dialog>
		);
	}

}

/* eslint-disable-next-line react/no-multi-comp */
const _DialogWithButtonGroup = (props, ref) => <DialogWithButtonGroup {...props} forwardedRef={ref} />;
_DialogWithButtonGroup.displayName = 'DialogWithButtonGroup';

// NOTE: Resolves an issue where enzyme@^3.3.0 and enzyme-adapter-react-16@^1.1.1 don't support `forwardRef`.
export default process.env.NODE_ENV === 'test' ? DialogWithButtonGroup : React.forwardRef(_DialogWithButtonGroup);
