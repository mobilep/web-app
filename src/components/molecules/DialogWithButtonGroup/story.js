import React, { Fragment } from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';

import DialogWithButtonGroup from './';
import { Button } from '../../atoms';

const dialogRef = React.createRef();

storiesOf('2. Molecules / DialogWithButtonGroup', module)
	.add('default', () => (
		<Fragment>
			<DialogWithButtonGroup
				actionButtonLabel={text('Action button label', 'Action')}
				cancelButtonLabel={text('Cancel button label', 'Cancel')}
				ref={dialogRef}
			>
				<h2>Dialog content is here.</h2>
			</DialogWithButtonGroup>
			<Button primary onClick={() => dialogRef.current.showModal()}>Open dialog</Button>
		</Fragment>
	));

