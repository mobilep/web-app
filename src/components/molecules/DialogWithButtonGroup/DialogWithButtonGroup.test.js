import React from 'react';
import { mount } from 'enzyme';
import mockMatchMedia from '../../../helpers/test/matchMedia';

import DialogWithButtonGroup from './DialogWithButtonGroup';

describe('Atom - DialogWithButtonGroup', () => {
	beforeAll(mockMatchMedia);

	it('Should render', () => {
		global.HTMLDialogElement = true;
		const wrapper = mount(
			<DialogWithButtonGroup
				actionButtonLabel='Action button'
				cancelButtonLabel='Cancel button'
			>
				DialogWithButtonGroup content
			</DialogWithButtonGroup>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
