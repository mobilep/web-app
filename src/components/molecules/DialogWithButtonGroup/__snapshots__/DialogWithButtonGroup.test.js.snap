// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - DialogWithButtonGroup Should render 1`] = `
<DialogWithButtonGroup
  actionButtonLabel="Action button"
  cancelButtonLabel="Cancel button"
  onActionButtonClick={[Function]}
  onCancelButtonClick={[Function]}
>
  <Dialog
    className="DialogWithButtonGroup"
    closeButton={true}
    onClose={[Function]}
  >
    <dialog
      className="Dialog-landscapeMode DialogWithButtonGroup"
    >
      <button
        className="Dialog-close-button"
        onClick={[Function]}
      >
        <CloseIcon>
          <SVGWrapper
            color="currentColor"
            height={31}
            width={31}
          >
            <svg
              className="svgWrapper"
              version="1.1"
              viewBox="0 0 31 31"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <g
                fill="none"
                fillRule="evenodd"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
              >
                <path
                  d="M21.333 21.333l-6.666-6.666L21.333 8"
                />
                <path
                  d="M8 21.333l6.667-6.666L8 8"
                />
              </g>
            </svg>
          </SVGWrapper>
        </CloseIcon>
      </button>
      <div
        className="DialogWithButtonGroup-content"
      >
        DialogWithButtonGroup content
      </div>
      <div
        className="DialogWithButtonGroup-buttonGroup"
      >
        <Button
          color="grey"
          onClick={[Function]}
        >
          <button
            className="Button Button-grey"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Cancel button
          </button>
        </Button>
        <Button
          onClick={[Function]}
        >
          <button
            className="Button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Action button
          </button>
        </Button>
      </div>
    </dialog>
  </Dialog>
</DialogWithButtonGroup>
`;
